experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -202,6 +205,66 @@ export abstract class LazyField<TKind extends FieldKindTypes, TTypes extends All
 		);
 	}
 
+	public reduce<U>(
+		callbackfn: (
+			previousValue: U,
+			currentValue: UnboxNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => U | Stop<U>,
+		initialValue?: U | undefined,
+	): UnboxNodeUnion<TTypes> | U {
+		return this.reduceImpl(
+			callbackfn,
+			(cursor) => unboxedUnion(this.context, this.schema, cursor),
+			initialValue,
+		);
+	}
+
+	public reduceBoxed<U>(
+		callbackfn: (
+			previousValue: U,
+			currentValue: TypedNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => U | Stop<U>,
+		initialValue?: U | undefined,
+	): TypedNodeUnion<TTypes> | U {
+		return this.reduceImpl(
+			callbackfn,
+			(cursor) => makeTree(this.context, cursor) as TypedNodeUnion<TTypes>,
+			initialValue,
+		);
+	}
+
+	private reduceImpl<Accumulation, T>(
+		callbackfn: (
+			previousValue: Accumulation,
+			currentValue: T,
+			currentIndex: number,
+			sequence: this,
+		) => Accumulation | Stop<Accumulation>,
+		getCurrentValue: (cursor: ITreeSubscriptionCursor) => T,
+		initialValue?: Accumulation,
+	): T | Accumulation {
+		let accumulation: Accumulation | undefined = initialValue;
+		forEachNode(this[cursorSymbol], (cursor) => {
+			const element = getCurrentValue(cursor);
+			const result =
+				accumulation !== undefined
```

Comment:
This is wrong in two cases:
1. When the caller to reduce explicitly passes in undefined as the initialValue.
2. When the callback on the previous item returned undefined .
For example if I use reduce to implement find, I might have my callback return `undefined|TreeNode`, and pass undefined as the initial value. Such an implementation of find will not work correctly.

Infact, I think anything with undefined initial value will never call the callback and just return undefined at the end. I think you need more tests.